cos-fleetshard-operator/src/main/java/org/bf2/cos/fleetshard/operator/connector/ConnectorController.java

```diff
@@ -393,6 +403,50 @@ private UpdateControl<ManagedConnector> handleAugmentation(ManagedConnector conn
             .withName(connector.getSpec().getDeployment().getSecret())
             .get();
 
+        ConfigMap configMap = kubernetesClient.configMaps()
+            .inNamespace(connector.getMetadata().getNamespace())
+            .withName(ConfigMaps.generateConnectorConfigMapId(connector.getSpec().getDeploymentId()))
+            .get();
+
+        if (configMap == null) {
+            LOGGER.info(
+                "Configmap not found (cluster_id: {}, namespace_id: {}, connector_id: {}, deployment_id: {}), creating a new one",
+                connector.getSpec().getClusterId(),
+                connector.getMetadata().getNamespace(),
+                connector.getSpec().getConnectorId(),
+                connector.getSpec().getDeploymentId());
+
+            configMap = new ConfigMap();
+            configMap.setMetadata(new ObjectMeta());
+            configMap.getMetadata().setNamespace(connector.getMetadata().getNamespace());
+            configMap.getMetadata().setName(ConfigMaps.generateConnectorConfigMapId(connector.getSpec().getDeploymentId()));
+
+            Resources.setLabels(
+                configMap,
+                LABEL_CLUSTER_ID, connector.getSpec().getClusterId(),
+                LABEL_CONNECTOR_ID, connector.getSpec().getConnectorId(),
+                LABEL_DEPLOYMENT_ID, connector.getSpec().getDeploymentId());
+
+            Resources.setOwnerReferences(
+                configMap,
+                connector);
+
+            // copy operator type
+            Resources.setLabel(
+                secret,
```

Comment:
Is this really supposed to be the `secret`, or should it be the `configMap`?
If it is the `configMap`, can it be done in L424? Also, I'd also suggest moving the secret query in L401 to after this `if` block.

Reply:
Ah! Good catch. This also highlights that some tests are missing.

Reply:
done